config/soap-client-notifications.php

```diff
@@ -0,0 +1,143 @@
+    (new Assembler\FluentSetterAssemblerOptions())
+        ->withTypeHints()
+        ->withReturnType()
+);
+
+$arraySetterAssembler = new OpenEuropa\Assembler\ArraySetterAssembler(
+    (new OpenEuropa\Assembler\ArraySetterAssemblerOptions())
+        ->whitelist($specialClassesAndProperties)
+);
+
+$defaultGetterAssembler = new Assembler\GetterAssembler(
+    (new Assembler\GetterAssemblerOptions())
+        ->withReturnType()
+        ->withBoolGetters()
+);
+
+$arrayGetterAssembler = new OpenEuropa\Assembler\ArrayGetterAssembler(
+    (new OpenEuropa\Assembler\ArrayGetterAssemblerOptions())
+        ->whitelist($specialClassesAndProperties)
+);
+
+$nullableGetterAssembler = new OpenEuropa\Assembler\NullableGetterAssembler(
+    (new OpenEuropa\Assembler\NullableGetterAssemblerOptions())
+        ->blacklist($specialClassesAndProperties)
+);
+
+$fluentAdderAssembler = new OpenEuropa\Assembler\FluentAdderAssembler(
+    (new OpenEuropa\Assembler\FluentAdderAssemblerOptions())
+        ->whitelist($specialClassesAndProperties)
+);
+
+$hasPropertyAssembler = new OpenEuropa\Assembler\HasPropertyAssembler();
+
+return Config::create()
+    ->setWsdl('resources/EpoetryNotificationReceiverService.xml')
+    ->setTypeDestination('src/Notification/Type')
+    ->setTypeNamespace('OpenEuropa\EPoetry\Notification\Type')
+    ->setClientDestination('src/Notification/')
+    ->setClientName('EPoetryNotificationClient')
+    ->setClientNamespace('OpenEuropa\EPoetryNotification')
+    ->setClassMapDestination('src/Notification/')
+    ->setClassMapName('EPoetryNotificationClassmap')
+    ->setClassMapNamespace('OpenEuropa\EPoetry\Notification')
+// We have to do this as the SOAP handler will erroneously create duplicate
+// public properties when a value object extends another one with those
+// same properties marked as "private".
+    ->setRuleSet(
+        new Rules\RuleSet(
+            [
+                new Rules\AssembleRule(new Assembler\ClassMapAssembler()),
+            ]
+        )
+    )
+    ->addRule(
+        new Rules\TypenameMatchesRule(
+            new Rules\AssembleRule(new Assembler\ResultAssembler()),
+            '/Response$/'
+        )
+    )
+// Generate SOAP request classes.
+//
+// Request objects must implement \Phpro\SoapClient\Type\RequestInterface
+// The rule matches the following SOAP types:
+//
+// - correctTranslation
+// - createRequests
+// - findLinguisticRequest
+// - getLinguisticRequest
+// - modifyRequest
+// - receiveNotifications
+    ->addRule(
+        new Rules\TypenameMatchesRule(
+            new Rules\AssembleRule(new Assembler\RequestAssembler()),
+            '/(CorrectTranslation|CreateRequests|ReceiveNotifications|Request)$/'
+        )
+    )
+    ->addRule(
+        new Rules\TypenameMatchesRule(
+            new Rules\AssembleRule(new Assembler\ExtendAssembler(DgtDocument::class)),
+            '/^(Auxiliary|Correction|Original)Document$/'
```

Comment:
Also don't know is this rule have any sense withing Notification client generator, Because we don't have any *Document